std/stdio.d

```diff
@@ -2919,7 +2933,7 @@ $(D Range) that locks the file and allows fast writing to it.
                     else
                     {
                         char[4] buf;
-                        immutable size = encode!(UseReplacementDchar.yes)(buf, c);
+                        immutable size = encode(buf, c);
```

Comment:
Maybe it's time for us to make another big push to kill autodecoding for good.  It's been almost universally hated by all, and while we did finally manage to convince Andrei the last time round that it's actually a bad thing (after many years of interminable arguments), we didn't go far enough, IMO -- we only got as far as introducing `byCodeUnit`, et al, and to try to reduce reliance on autodecoding in various Phobos functions.

But it has been an ongoing source of performance concerns, and now `nothrow` and `@nogc` unfriendliness, that perhaps we can make a very strong case for getting rid of it forever.  I, for one, plan to celebrate the day the last of autodecoding is finally killed off from Phobos code forever.

Reply:
@quickfur Did you see this comment https://github.com/dlang/phobos/pull/6073#issuecomment-360833216 ?

Reply:
RCStr has been talked about since, what, a year ago or more?  When is it ever going to materialize? What's the holdup?

Reply:
@quickfur DIP1000. I stalled Walter's PRs for it because I believed it wasn't handling GC-ed objects correctly (and Martin agreed https://github.com/dlang/phobos/pull/5052#issuecomment-338697155). There's been no work on DIP1000 since my comments on the matter so I think Walter dropped it for some reason.

Reply:
Here's a list of what modules aren't compatible with DIP1000 as of now: https://github.com/dlang/phobos/projects/6

There has been some recent work on DIP1000 at https://github.com/dlang/phobos/pull/5915 and https://github.com/dlang/phobos/pull/6041. The first one stalled because DIP1000 uses a different mangling :/

Reply:
>There has been some recent work on DIP1000 at #5915 and #6041

Ok, there's been no work *by Walter* since

Reply:
While dip1000 isn't perfect by any stretch, I still would welcome it as an improvement on the current situation.